### PR TITLE
fix(zip): fixed zip function

### LINKED ISSUE #6: Fix zip function
`zip` function with the input: 

`[1,2,3,4,5] [9,8]`

gives the following output: 

`[[ 1, 9 ], [ 2, 8 ], [ 3, undefined ], [ 4, undefined ], [ 5, undefined ]]`

instead of

`[[1,9],[2,8]]`

### PR DESCRIPTION
**Pull request by AngyDev**
"fix #6"

### CODE REVIEW

File: src/array/zip/index.ts [hunk 1]
```diff
@@ -1,5 +1,9 @@
 export function zip(arrays: any[][]): any[][] {
```
Comment (micheleriva) [inline src/array/zip/index.ts]: Thanks for your PR! I just noticed an error I made when implementing the original `zip` method. Could you please refactor the signature from:

```ts
export function zip(arrays: any[][]): any[][]
```

into:

```ts
export function zip<T>(arrays: T[][]): T[][]
```

that way we can specify the input and return types of a `zip` function 🙂

**Review by micheleriva — APPROVED** [review]

File: src/array/zip/index.ts (hunk 1 above)
Comment (AngyDev) [reply src/array/zip/index.ts]: Thank you. The same refactor is also for the inherit function, right?

Comment (micheleriva) [reply src/array/zip/index.ts]: Yes thanks!

**Comment by AngyDev** [comment]
With the refactor of the signature we have a problem with the tests because if I pass a different type of array I have an error that I can't assign a different type, maybe was `any` the right one?

**Comment by AngyDev** [comment]
I understood the correct way to correct the test with the changes in the signature of the zip function

**Comment by micheleriva** [comment]
@AngyDev awesome, thank you so much!

**Comment by AngyDev** [comment]
@micheleriva thank you!! It is a pleasure!

**Pull request merged**